src/main/scala/uk/gov/hmrc/play/http/HeaderNames.scala

```diff
@@ -20,20 +20,46 @@ object HeaderNames {
 
   import play.api.http.HeaderNames.AUTHORIZATION
 
-  val authorisation = AUTHORIZATION
-  val xForwardedFor = "x-forwarded-for"
-  val xRequestId = "X-Request-ID"
-  val xRequestTimestamp = "X-Request-Timestamp"
-  val xSessionId = "X-Session-ID"
-  val xRequestChain = "X-Request-Chain"
-  val trueClientIp = "True-Client-IP"
-  val trueClientPort = "True-Client-Port"
-  val token = "token"
-  val surrogate = "Surrogate"
-  val otacAuthorization = "Otac-Authorization"
-  val googleAnalyticTokenId = "ga-token"
-  val googleAnalyticUserId  = "ga-user-cookie-id"
-  val deviceID  = "deviceID" // not a typo, should be ID
+  /*
+   * this isn't ideal, but downstream apps still want to refer to typed header values
```

Comment:
It's a map because HeaderNames, prior to the change, was a collection of key->value pairs, i.e. not tuples because the keys are unique. We could use a sequence with index based access but, personally, I thought that was a little less clear (and also broken by "accidental" re-ordering of the collection). I think case objects etc would be overkill for something this simple where all we wanted to do was to be able to retrieve a key set whilst maintaining the val references for backward compatibility with existing client code which might need them.